paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/PaymentMethodVEmbeddedLayoutUI.kt

```diff
@@ -0,0 +1,210 @@
+    savedPaymentMethodAction: PaymentMethodVerticalLayoutInteractor.SavedPaymentMethodAction,
+    selection: PaymentSelection?,
+    isEnabled: Boolean,
+    onViewMorePaymentMethods: () -> Unit,
+    onManageOneSavedPaymentMethod: (DisplayableSavedPaymentMethod) -> Unit,
+    onEditPaymentMethod: (DisplayableSavedPaymentMethod) -> Unit,
+    onSelectSavedPaymentMethod: (DisplayableSavedPaymentMethod) -> Unit,
+    imageLoader: StripeImageLoader,
+    rowStyle: Embedded.RowStyle,
+    modifier: Modifier = Modifier,
+) {
+    val arrangement = if (rowStyle is Embedded.RowStyle.FloatingButton) {
+        Arrangement.spacedBy(rowStyle.spacingDp.dp)
+    } else {
+        Arrangement.Top
+    }
+    Column(modifier = modifier, verticalArrangement = arrangement) {
+        if (rowStyle.topSeparatorEnabled()) OptionalEmbeddedDivider(rowStyle)
+        if (displayedSavedPaymentMethod != null) {
+            SavedPaymentMethodRowButton(
+                displayableSavedPaymentMethod = displayedSavedPaymentMethod,
+                isEnabled = isEnabled,
+                isSelected = selection?.isSaved == true,
+                trailingContent = {
+                    SavedPaymentMethodTrailingContent(
+                        displayedSavedPaymentMethod = displayedSavedPaymentMethod,
+                        savedPaymentMethodAction = savedPaymentMethodAction,
+                        onViewMorePaymentMethods = onViewMorePaymentMethods,
+                        onEditPaymentMethod = onEditPaymentMethod,
+                        onManageOneSavedPaymentMethod = { onManageOneSavedPaymentMethod(displayedSavedPaymentMethod) },
+                    )
+                },
+                onClick = { onSelectSavedPaymentMethod(displayedSavedPaymentMethod) },
+                rowStyle = rowStyle
+            )
+        }
+
+        OptionalEmbeddedDivider(rowStyle)
+
+        val selectedIndex = remember(selection, paymentMethods) {
+            if (selection == null || selection.isSaved) {
+                -1
+            } else {
+                val code = selection.code()
+                paymentMethods.indexOfFirst { it.code == code }
+            }
+        }
+
+        paymentMethods.forEachIndexed { index, item ->
+            NewPaymentMethodRowButton(
+                isEnabled = isEnabled,
+                isSelected = index == selectedIndex,
+                displayablePaymentMethod = item,
+                imageLoader = imageLoader,
+                rowStyle = rowStyle
+            )
+
+            if (index != paymentMethods.lastIndex) {
+                OptionalEmbeddedDivider(rowStyle)
+            } else if (rowStyle.bottomSeparatorEnabled()) {
+                OptionalEmbeddedDivider(rowStyle)
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalEmbeddedPaymentElementApi::class)
+@Composable
+private fun OptionalEmbeddedDivider(rowStyle: Embedded.RowStyle) {
+    if (rowStyle !is Embedded.RowStyle.FloatingButton) {
```

Comment:
Ideally we don't need an is check here. Can we add something to the row style to make this an easy conditional defined at the row style level?

Reply:
Add an `internal abstract fun hasSeparators()` to `RowStyle`